extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeMixPlaylistExtractor.java

```diff
@@ -0,0 +1,182 @@
+import java.io.IOException;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import org.schabi.newpipe.extractor.StreamingService;
+import org.schabi.newpipe.extractor.downloader.Downloader;
+import org.schabi.newpipe.extractor.exceptions.ExtractionException;
+import org.schabi.newpipe.extractor.exceptions.ParsingException;
+import org.schabi.newpipe.extractor.linkhandler.ListLinkHandler;
+import org.schabi.newpipe.extractor.localization.TimeAgoParser;
+import org.schabi.newpipe.extractor.playlist.PlaylistExtractor;
+import org.schabi.newpipe.extractor.stream.StreamInfoItem;
+import org.schabi.newpipe.extractor.stream.StreamInfoItemsCollector;
+
+/**
+ * A YoutubePlaylistExtractor for a mix (auto-generated playlist). It handles urls in the format of
+ * "youtube.com/watch?v=videoId&list=playlistId"
+ */
+public class YoutubeMixPlaylistExtractor extends PlaylistExtractor {
+
+    private final static String CONTENTS = "contents";
+    private final static String RESPONSE = "response";
+    private final static String PLAYLIST = "playlist";
+    private final static String TWO_COLUMNS_WATCH_NEXT_RESULTS = "twoColumnWatchNextResults";
+    private final static String PLAYLIST_PANEL_VIDEO_RENDERER = "playlistPanelVideoRenderer";
+
+    private JsonObject playlistData;
+
+    public YoutubeMixPlaylistExtractor(StreamingService service, ListLinkHandler linkHandler) {
+        super(service, linkHandler);
+    }
+
+    @Override
+    public void onFetchPage(@Nonnull Downloader downloader)
+        throws IOException, ExtractionException {
+        final String url = getUrl() + "&pbj=1";
+        final JsonArray ajaxJson = getJsonResponse(url, getExtractorLocalization());
+        JsonObject initialData = ajaxJson.getObject(3).getObject(RESPONSE);
+        try {
+            playlistData = initialData.getObject(CONTENTS).getObject(TWO_COLUMNS_WATCH_NEXT_RESULTS)
+                .getObject(PLAYLIST).getObject(PLAYLIST);
+        } catch (NullPointerException e) {
+            throw new ExtractionException(e);
+        }
+
+    }
+
+    @Nonnull
+    @Override
+    public String getName() throws ParsingException {
+        try {
+            final String name = playlistData.getString("title");
+            if (name != null) {
+                return name;
+            } else {
+                return "";
+            }
+        } catch (Exception e) {
+            throw new ParsingException("Could not get playlist name", e);
+        }
+    }
+
+    @Override
+    public String getThumbnailUrl() throws ParsingException {
+        try {
+            final String playlistId = playlistData.getString("playlistId");
+            final String videoId;
+            if (playlistId.startsWith("RDMM")) {
+                videoId = playlistId.substring(4);
+            } else {
+                videoId = playlistId.substring(2);
+            }
+            if (videoId.isEmpty()) {
+                throw new ParsingException("");
+            }
```

Comment:
I would move this piece of code into `getThumbnailUrlFromId`, and maybe throw a ParsingException with a description like "videoId is empty".